internal/service/employee/employee_default.go

```diff
@@ -73,11 +81,8 @@ func (s *service) Update(id int, employee models.Employee) (models.Employee, err
 }
 
 func (s *service) Delete(id int) error {
-	_, ok := s.repository.GetById(id)
-	if !ok {
-		return fmt.Errorf("%w : employee with id %d not found", errors.ErrNotFound, id)
+	if err := s.repository.Delete(id); err != nil {
+		return fmt.Errorf("%w : failed to delete employee with id %d", errors.ErrNotFound, id)
```

Comment:
En el repo ya se valida si es un not found cuando se verifica si rowsAffected = 0

Desde el service solo se debe retornar el error
```suggestion
	if err := s.repository.Delete(id); err != nil {
		return err
	}

	return nil
```